File: Native/Test/Node/CORE-150.js
/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

FABRIC = require('Fabric').createClient();

op = FABRIC.DependencyGraph.createOperator( "op" );
op.setEntryPoint('entry');
op.setSourceCode("operator entry( io Scalar input[][] ) {}");

binding = FABRIC.DependencyGraph.createBinding();
binding.setOperator( op );
binding.setParameterLayout( [ "self.input" ] );

node = FABRIC.DependencyGraph.createNode( "node" );
node.addMember( "input", "Scalar[]" );
node.bindings.append( binding );
node.setData( "input", [17] );
console.log( node.getErrors() );

FABRIC.close();

File: Native/Test/Python/CORE-150.py
#
#  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
#

import fabric
fabricClient = fabric.createClient()

op = fabricClient.DependencyGraph.createOperator( "op" )
op.setEntryPoint('entry')
op.setSourceCode("operator entry( io Scalar input[][] ) {}")

binding = fabricClient.DependencyGraph.createBinding()
binding.setOperator( op )
binding.setParameterLayout( [ "self.input" ] )

node = fabricClient.DependencyGraph.createNode( "node" )
node.addMember( "input", "Scalar[]" )
node.bindings.append( binding )
node.setData( "input", [17] )
print( node.getErrors() )

fabricClient.close()
